mantidproject/mantid
# Adding alternative inputs to select old instrument definition files

### Description of work
With DGS instruments implementing upgrades, new detectors will be added and detector positions will change. This will result in creating multiple detector .xml files to reflect new changes. But older definition files are needed when examining older data. An alternative input option have been implemented to address this issue.

EWM ref: https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=342

#### Summary of work

A new "InstrumentDefinitionFIle" option has been added in the instrument selection drop down menu, when this option is selected, a new field called InstrumentDefinitionFile will appear. Use can browse the correct IDF file location and choose to use this to generate gropuing files. This option is only suitable for DGS instruments at SNS ("ARCS", "SEQUOIA", "HYSPEC, "CNCS").





#### Further detail of work


### To test:


Open mantid, load "GenerateGroupingSNSInelastic" script, in instrument drop down menu select "InstrumentDefinitionFile", a new field should appear that allows tester to specify which IDF file to use. Only IDF files associated with "ARCS", "SEQUOIA", "HYSPEC, "CNCS" can be selected, otherwise an error will appear "Select the instrument definition file from only one of ARCS, SEQUOIA, CNCS, HYSPEC".





---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
